### DISCUSSION TITLE
Reusing ISR cache across builds on Vercel

### DISCUSSION THREAD

**Discussion by rmanalan** [vercel/next.js#32266] [Help] [answered]
https://vercel.com/docs/concepts/next.js/incremental-static-regeneration#isr:-not-just-caching states:

> ISR is designed to persist your generated pages between deployments. This means you are able to rollback instantly and not lose your previously generated pages.

That makes sense to me. However, I'm wondering if ISR+Vercel utilizes the previous build's page cache if the new page generation is turned off at build time (i.e., `getStaticPaths()` returns `{ paths: [], fallback: 'blocking' }`? OR... will Next.js think that the page doesn't exist in the cache then generate it on demand?

I'm trying to see if it's possible to set up ISR so that it only ever regenerates the page if the props have actually changed -- and not on every build.

**Comment by drpancake** [comment]
I'm curious about this too. It would be great if the cache could persist ISR-rendered pages between deploys and served up the cached page if that page's components have not been modified at all.

In my own testing I can see that from a user's perspective the cache gets wiped on each deploy.

**Comment by Hanskrogh** [comment]
Any news on this?

**Comment by leerob** [comment, marked as answer]
This is now possible, learn more here: https://vercel.com/docs/infrastructure/data-cache

**Comment by victorcarvalhosp** [comment]
Does this ISR cache between builds works when using the page router? For me it's always cleaning up after a new build.

**Reply by sigginjals** [reply]
@victorcarvalhosp I'm experiencing the same behaviour, did you figure out a way to get SSG to persist between deployments?

**Reply by fxmb** [reply]
Has anyone managed this? We are using the app router and not sure how and if this works?
